web/ui_utils.js

```diff
@@ -355,22 +355,25 @@ function noContextMenuHandler(e) {
 
 /**
  * Returns the filename or guessed filename from the url (see issue 3455).
- * url {String} The original PDF location.
- * defaultFilename {string} The value to return if the file name is unknown.
- * @return {String} Guessed PDF file name.
+ * @param {string} url - The original PDF location.
+ * @param {string} defaultFilename - The value returned if the filename is
+ *   unknown, or the protocol is unsupported.
+ * @returns {string} Guessed PDF filename.
  */
-function getPDFFileNameFromURL(url, defaultFilename) {
-  if (typeof defaultFilename === 'undefined') {
-    defaultFilename = 'document.pdf';
+function getPDFFileNameFromURL(url, defaultFilename = 'document.pdf') {
+  if (url.indexOf('data:') === 0) {
```

Comment:
Thank you for the review!
I've made the requested changes, and also added a few more unit-tests for the (whitespace) case.